ExampleMod/ExamplePlayer.cs

```diff
@@ -690,6 +690,77 @@ public class ExamplePlayer : ModPlayer
 			layers.Insert(0, MiscEffectsBack);
 			MiscEffects.visible = true;
 			layers.Add(MiscEffects);
+
+			/* This shows an example of how we can give the held sprite of an item a glowmask or animation. This example goes over drawing a mask for a specific item, but you may want to
+			 * expand your layer to cover multiple items, rather than adding in a new layer for every single item you want to create a mask for.
+			 * 
+			 * PlayerLayers have many other uses, basically anything you would want to visually create on a player, such as a custom accessory layer, holding a weapon, or anything else you can think of
+			 * that would involve drawing sprites on a player or their held items. these examples only serve to illustrate common requests and can be extrapolated into anything you want. DrawData's constructor has
+			 * the same set of params as SpriteBatch.Draw(), so you can get as creative as you want here.
+			 * 
+			 * Note that if you want to give your held item an animation, you should set that item's NoUseGraphic field to true so that the entire spritesheet for that item wont draw.
+			 */
+
+			//This layer is for our glowing sword example!
+			Action<PlayerDrawInfo> layerTarget = s => DrawSwordGlowmask(s); //the Action<T> of our layer. This is the delegate which will actually do the drawing of the layer.
+			PlayerLayer layer = new PlayerLayer("ExampleMod", "Sword Glowmask", layerTarget); //Instantiate a new instance of PlayerLayer to insert into the list
+			layers.Insert(layers.IndexOf(layers.FirstOrDefault(n => n.Name == "Arms")), layer); //Insert the layer at the appropriate index. 
+																								//the "Arms" layer is directly above the "HeldItem" layer so our layer will appear below the players hand and above their item.
+
+			//this layer is for our animated sword example! this is pretty much the same as the above layer insertion.
+			Action<PlayerDrawInfo> layerTarget2 = s => DrawSwordAnimation(s);
+			PlayerLayer layer2 = new PlayerLayer("ExampleMod", "Sword Animation", layerTarget2);
+			layers.Insert(layers.IndexOf(layers.FirstOrDefault(n => n.Name == "Arms")), layer2);
+		}
+		private void DrawSwordGlowmask(PlayerDrawInfo info)
+		{
+			Player player = info.drawPlayer; //the player!
+
+			if (player.HeldItem.type == ModContent.ItemType<Items.Weapons.ExampleGlowSword>() && player.itemAnimation != 0) //We want to make sure that our layer only draws when the player is swinging our specific item.
```

Comment:
Since PlayerLayers are already quite OO, I'd like to suggest a couple of improvements:
Abstract both Draw methods into one generic version in which you pass PlayerDrawInfo, the Texture, frame and Color.
Call it something like `ItemDrawLayer(PlayerDrawInfo info, Texture2D texture, Rectangle? frame, Color color)` because it's drawlayer for held item. This shows proper re-use of the method since all of their logic is the same.